kernels/optimized/cpu/op_linear.cpp

```diff
@@ -56,6 +50,25 @@ Tensor& opt_linear_out(
         size_t k = in.sizes()[in.dim() - 1];
         size_t m = mat2.size(0);
 
+        // If bias is provided, verify its shape and pre-fill the output tensor.
+        if (bias.has_value()) {
+          auto bias_value = bias.value();
+          // Check that bias is 1D and its size matches m.
+          ET_KERNEL_CHECK(ctx, bias_value.dim() == 1 && bias_value.size(0) == m, InvalidArgument, out);
+          auto bias_ptr = bias_value.const_data_ptr<CTYPE>();
+          CTYPE* out_ptr = out.mutable_data_ptr<CTYPE>();
+          // Broadcast the bias to every column of the output.
+          for (size_t col = 0; col < n; ++col) {
```

Comment:
use c10::irange. you'll need to include the header and add a buck dep. -- `for (const auto col : c10::irange(n)) {`